app/assets/javascripts/Components/graders_manager.jsx

```diff
@@ -105,15 +127,38 @@ class GradersManager extends React.Component {
         skip_empty_submissions: this.state.skip_empty_submissions,
         groupings: groups,
         criteria: criteria,
-        graders: graders,
+        graders: Object.keys(groupsPerGrader).sort(
+          (a, b) => groupsPerGrader[a] - groupsPerGrader[b]
+        ),
+        groups_per_grader: Object.values(groupsPerGrader).sort((a, b) => a - b),
       },
     }).then(this.fetchData);
   };
 
+  calculateGroupsPerGrader = (graders, groups) => {
```

Comment:
This type of calculation work is normally done in the back-end, so you should move it to the main model method that's doing the heavy lifting.